### ISSUE TITLE
Drush 9 Support with BLT Doctor

### ISSUE THREAD

**Issue by aweingarten** [acquia/blt#2214]
My system information:
* BLT version: 9.x

Output of `blt doctor`:
```
+-----------------------------+----------------------------------------------------------------------------------+
| Property                    | Value                                                                            |
+-----------------------------+----------------------------------------------------------------------------------+
| %paths                      |                                                                                  |
|  - %root                    | /var/www/mysite/docroot/../docroot                                                 |
|  - %site                    | sites/mysite.com                                                       |
|  - %modules                 | sites/all/modules                                                                |
|  - %themes                  | sites/all/themes                                                                 |
|  - %config-sync             | /var/www/mysite/config/mysite.com                                        |
|  - %files                   | sites/mysite.com/files                                                 |
|  - %temp                    | /tmp                                                                             |
|  - %private                 | /var/www/mysite/files-private                                                      |
| admin-theme                 | seven                                                                            |
| blt-version                 | 9.x-dev                                                                          |
| bootstrap                   | Successful                                                                       |
| config-sync                 | /var/www/mysite/config/mysite.com                                        |
| db-driver                   | mysql                                                                            |
| db-hostname                 | localhost                                                                        |
| db-name                     | drupal                                                                           |
| db-password                 | drupal                                                                           |
| db-port                     | 3306                                                                             |
| db-status                   | Connected                                                                        |
| db-username                 | drupal                                                                           |
| dev-desktop-enabled         | false                                                                            |
| drupal-settings-file        | sites/mysite.com/settings.php                                          |
| drupal-version              | 8.4.0                                                                            |
| drupal-vm-enabled           | true                                                                             |
| drush-alias-files           |                                                                                  |
| drush-conf                  | /var/www/mysite/vendor/drush/drush/drush.yml                                       |
|                             | /var/www/mysite/drush/drush.yml                                                    |
| drush-script                | /var/www/mysite/vendor/bin/drush                                                   |
| drush-temp                  | /tmp                                                                             |
| drush-version               | 9.0.0-beta7                                                                      |
| files                       | sites/mysite.com/files                                                 |
| install-profile             | myprofile                                                                              |
| local-drushrc               | /var/www/mysite/docroot/../docroot/sites/mysite.com/local.drushrc.php    |
| local-settings              | /var/www/mysite/docroot/../docroot/sites/mysite.com/settings/local.setti |
|                             | ngs.php                                                                          |
| modules                     | sites/all/modules                                                                |
| php-bin                     | /usr/bin/php7.1                                                                  |
| php-conf                    |                                                                                  |
|  - /etc/php/7.1/cli/php.ini | /etc/php/7.1/cli/php.ini                                                         |
| php-mysql                   | /var/run/mysqld/mysqld.sock                                                      |
| php-os                      | Linux                                                                            |
| private                     | /var/www/mysite/files-private                                                      |
| root                        | /var/www/mysite/docroot/../docroot                                                 |
| shell                       |                                                                                  |
| site                        | sites/mysite.com                                                       |
| temp                        | /tmp                                                                             |
| theme                       | freerange                                                                        |
| themes                      | sites/all/themes                                                                 |
| uri                         | default                                                                          |
+-----------------------------+----------------------------------------------------------------------------------+
+--------------------------+--------------------------------------------------------------------------------+
| Check                    | Outcome                                                                        |
+--------------------------+--------------------------------------------------------------------------------+
| checkDocrootExists       | Found docroot.                                                                 |
| checkCoreExists          | Drupal core exists                                                             |
| checkSettingsFile        | BLT settings are included in settings file.                                    |
| checkLocalSettingsFile   | Found your local settings file.                                                |
| checkLocalDrushFile      | Found your local drush settings file.                                          |
| checkUri                 | Site URI is not set                                                            |
|                          |                                                                                |
|                          | Is $options['uri'] set correctly in                                            |
|                          | /var/www/mysite/docroot/../docroot/sites/mysite.com/local.drushrc.php? |
|                          |                                                                                |
| checkFileSystem:%files   | Public files directory is writable.                                            |
| checkFileSystem:%private | Private files directory is writable.                                           |
| checkFileSystem:%temp    | Temporary files directory is writable.                                         |
| checkDbConnection        | Connected to database.                                                         |
| checkDrupalBootstrapped  | Bootstrapped Drupal via drush.                                                 |
| checkDrupalInstalled     | Drupal is installed.                                                           |
| checkCaching:page        | Drupal cache is disabled.                                                      |
| checkCaching:css         | CSS preprocessing is disabled.                                                 |
| checkCaching:js          | JS preprocessing is disabled.                                                  |
| checkCiConfig            | Git remotes are set in project.yml.                                            |
| checkComposer:require    | acquia/blt is in composer.json's require object.                               |
| checkBehat:exists        | Behat local settings file exists.                                              |
| checkBehat:root          | Behat drupal_root is set correctly for Drupal VM.                              |
| checkBehat:uri           | base_url in tests/behat/local.yml does not match the site URI.                 |
|                          |   Behat base_url is set to "http://local8.mysite.com".               |
|                          |   Drush site URI is set to default.                                            |
| checkProjectYml:keys     | project.yml has no deprecated keys.                                            |
| checkDrushAliases        | drush alias file does not exist!                                               |
|                          |                                                                                |
|                          | Create                                                                         |
|                          | /var/www/mysite/docroot/../docroot/../drush/site-aliases/aliases.drushrc.php     |
| checkDrupalVm            | Drupal VM is configured correctly.                                             |
| checkPhpDateTimezone     | PHP setting for date.timezone is correctly set                                 |
+--------------------------+--------------------------------------------------------------------------------+
```

I think that we need to refactor `blt doctor` to look for the new drush YML files.

**Comment by danepowell** [comment]
Could be related: #2213

**Comment by danepowell** [comment]
I'm going to keep this open until Drush beta9 drops, which will include a necessary fix for non-interactive commands (i.e. `drush sa` run in the VM)

**Comment by alex-moreno** [comment]
Is it OK to have client names here?

**Comment by danepowell** [comment]
@alex-moreno generally best not to disclose client names

**Comment by alex-moreno** [comment]
I ask because of the description, there are some references there

**Comment by danepowell** [comment]
@aweingarten ^

**Comment by danepowell** [comment]
Looks like this didn't make it into beta9 :cry: 

This is the patch we're waiting for:
https://github.com/drush-ops/drush/pull/3136

**Comment by danepowell** [comment]
While the linked issue is still open, I'm not sure it's actually blocking anything critical here.

Improvements have been made to BLT doctor and alias detection as part of other issues.

@aweingarten can you check if any action is still required here?

**Comment by grasmash** [comment]
Fixed in #2400.
